app/controllers/CanPayWithinSixMonthsController.scala

```diff
@@ -47,23 +48,28 @@ class CanPayWithinSixMonthsController @Inject() (
 
   import requestSupport._
 
-  val canPayWithinSixMonths: Action[AnyContent] = as.authenticatedJourneyAction { implicit request =>
-    request.journey match {
-      case j: Journey.BeforeRetrievedAffordabilityResult =>
-        JourneyIncorrectStateRouter.logErrorAndRouteToDefaultPage(j)
-
-      case j: Journey.AfterRetrievedAffordabilityResult =>
-        finalStateCheck(
-          j,
-          {
-            val previousAnswers = existingAnswersInJourney(request.journey)
-            val form = previousAnswers.fold(CanPayWithinSixMonthsForm.form)(value =>
-              CanPayWithinSixMonthsForm.form.fill(CanPayWithinSixMonthsFormValue.canPayWithinSixMonthsToFormValue(value)))
-
-            Ok(views.canPayWithinSixMonthsPage(form, remainingAmountToPay(j)))
-          }
-        )
+  def canPayWithinSixMonths(@unused regime: TaxRegime): Action[AnyContent] =
+    as.continueToSameEndpointAuthenticatedJourneyAction { implicit request =>
+      request.journey match {
+        case j: Journey.BeforeRetrievedAffordabilityResult =>
+          JourneyIncorrectStateRouter.logErrorAndRouteToDefaultPage(j)
+
+        case j: Journey.AfterRetrievedAffordabilityResult =>
+          finalStateCheck(
+            j,
+            {
+              val previousAnswers = existingAnswersInJourney(request.journey)
+              val form = previousAnswers.fold(CanPayWithinSixMonthsForm.form)(value =>
+                CanPayWithinSixMonthsForm.form.fill(CanPayWithinSixMonthsFormValue.canPayWithinSixMonthsToFormValue(value)))
+
+              Ok(views.canPayWithinSixMonthsPage(form, remainingAmountToPay(j)))
+            }
+          )
+      }
     }
+
+  def backFromPegaLanding(regime: TaxRegime): Action[AnyContent] = as.authenticatedJourneyAction { _ =>
```

Comment:
you shouldn't need this endpoint I think - they would just point directly to the `canPayWithinSixMonths` endpoint

Reply:
oh wait, I think I see what this is now. This should go in the test-only pega controller since we don't want to expose this in the actual service